Separating dashboard and outline

megoth opened the pull request:
No reason to reuse elements from the outline container that is used for displaying panes for a given subject.

This commit separates the logic for the two, and handles switching between them.

Not done any styling on cancel button, as I want to know how Tim wants to have it first.

This fixes https://github.com/solid/solid-panes/issues/154

This has some breaking changes in API that [mashlib](https://github.com/solid/mashlib/) uses, and mashlib will require these commit to continue functioning.

megoth committed:
Separating dashboard and outline

No reason to reuse elements from the outline container that is used for displaying panes for a given subject.

This commit separates the logic for the two, and handles switching between them.

Not done any styling on cancel button, as I want to know how Tim wants to have it first.

This fixes https://github.com/solid/solid-panes/issues/154

Vinnl commented on outline/manager.js:
```diff
@@ -365,19 +368,73 @@ module.exports = function (doc) {
-      // currentPane.dispatchEvent(new Event('clglobalAppTabsick'))
+  /**
+   *
+   * @param {HTMLElement} container The element that the dashboard should append dashboard to
```
This param seems to have been replaced by `getDashboardContainer`?

```suggestion
```

Vinnl commented on outline/manager.js:
```diff
@@ -365,19 +368,73 @@ module.exports = function (doc) {
+      outlineContainer.style.display = 'none'
+      dashboardContainer.style.display = 'inherit'
+      const tab = dashboardContainer.querySelector(`[data-name="${options.pane}"]`) ||
+        dashboardContainer.querySelector('[data-name]')
```
I presume `data-name` is already used somewhere? If not, it's a bit odd to have a data attribute with the same name as a regular attribute (i.e. `name`), and something like `data-global-pane-name` or something would be clear. But no biggy if it's already used.

megoth replied:
You're correct, I should use a more specific name than one that resembles standard attribute `name`. Will use `data-global-pane-name` instead.

Vinnl reviewed (approved):
Overal looks good; fine with merging if the JSDoc is fixed.

megoth committed:
Fixing up a bit on JSDocs parts related to latest changes

megoth committed:
Making use of new option onClose in UI.widgets.tabs

This presents a cancelButton next to the tabs.

This relies on changes in https://github.com/solid/solid-ui/pull/110

megoth committed:
Renaming data-attribute per Vincent's suggestion

timbl commented on outline/manager.js:
```diff
@@ -365,19 +372,80 @@ module.exports = function (doc) {
+        return
+      }
+      console.warn('Found no tabs in global dashboard to open')
+      dashboardContainer.innerHTML = ''
```
If no specific tab is specified by the caller, why not let the tabs widget just select the default tab?  Doesn't `dashboardContainer.innerHTML = ''` zap the dashboard?

megoth replied:
It was to catch the case where a user tries to open a dashboard pane that isn't available - it should not happen, so was just to fetch that error.

But I've updated with a commit that tries to make this clearer. And yes, the fallback is that tabs will open the first one, so removed some redundant logic in mashlib code.

megoth committed:
Making logic on which tab to open a bit clearer to read

Based on Tims feedback - https://github.com/solid/solid-panes/pull/166#discussion_r318759500
